### DISCUSSION TITLE
Brainstorm: A note taking app with typst, how hard would it be?

### DISCUSSION THREAD

**Discussion by glyh** [typst/typst#4275] [Q&A]
So after spending a lot of time on logseq, org-mode and neorg(on neovim), I'm not happy with any of them. org-mode is almost bearable for me but emacs eco-system seems so broken. I would like to create a new frontend for neovim or another editor(if I ever have the time, I know this is a lot of work). 

Generally I expect:
- WYSIWYG, typora like experience
- Links that organize your note in an org-roam like behavior
- Instaneous editing feedback

I'm not expecting the maintainer to implement this, but I'm just generally curious: how hard would it be to implement these given the current typst architecture?

Thanks!

**Comment by Enter-tainer** [comment]
> WYSIWYG, typora like experience

very hard

**Reply by glyh** [reply]
I am aware for some constructs that is not possible at all. But I think probably for bare minium support(text, tables) this could be a thing? Looks like for consistency's sake there is not much syntax sugar given by typst though. Maybe a layer on top will be needed.

**Comment by TS60** [comment]
I think it would be possible to achieve by using something like incremental compilation and compiling to svg. Then in realtime print the new svg in the browser, compare the last svg with the new one on each `k` text input changes. This would require a good algorithm with constant runtime to detect where the changes were made. But it would only work good when SVG supports real text and not outlined text.

From the GUI, supply some input methods like tables and images with approximate fitting sizes, typst will return the figures with ca. 80 % expectation, than make it possible to change the sizes, typst will again return a new size, which now fits 90 % expectation and so on.

This is the best you can get.
